integrated_channels/blackboard/models.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+"""
+Database models for Enterprise Integrated Channel Blackboard.
+"""
+
+from simple_history.models import HistoricalRecords
+
+from django.db import models
+from django.utils.encoding import python_2_unicode_compatible
+
+from integrated_channels.integrated_channel.models import EnterpriseCustomerPluginConfiguration
+from .transmitters.content_metadata import BlackboardContentMetadataTransmitter
+from .exporters.content_metadata import BlackboardContentMetadataExporter
+
+
+@python_2_unicode_compatible
+class BlackboardEnterpriseCustomerConfiguration(EnterpriseCustomerPluginConfiguration):
+    """
+    The Enterprise-specific configuration we need for integrating with Blackboard.
+
+    .. no_pii:
+    """
+
+    client_id = models.CharField(
+        max_length=255,
+        null=True,
+        verbose_name="API Client ID",
+        help_text=(
+            "The API Client ID provided to edX by the enterprise customer to be used to make API "
+            "calls on behalf of the customer."
+        )
+    )
+
+    client_secret = models.CharField(
+        max_length=255,
+        null=True,
+        verbose_name="API Client Secret",
+        help_text=(
+            "The API Client Secret provided to edX by the enterprise customer to be used to make "
+            " API calls on behalf of the customer."
+        )
+    )
+
+    blackboard_base_url = models.CharField(
+        max_length=255,
+        null=True,
+        verbose_name="Base URL",
+        help_text="The base URL used for API requests to Blackboard, i.e. https://blackboard.com."
+    )
+
+    refresh_token = models.CharField(
+        max_length=255,
+        blank=True,
+        verbose_name="Oauth2 Refresh Token",
+        help_text="The refresh token provided by Blackboard along with the access token request,"
+                  "used to re-request the access tokens over multiple client sessions."
+    )
+
+    history = HistoricalRecords()
+
+    class Meta:
+        app_label = 'blackboard'
+
+    def __str__(self):
+        """
+        Return human-readable string representation.
+        """
+        return "<BlackboardEnterpriseCustomerConfiguration for Enterprise {enterprise_name}>".format(
+            enterprise_name=self.enterprise_customer.name
+        )
+
+    def __repr__(self):
+        """
+        Return uniquely identifying string representation.
+        """
+        return self.__str__()
+
+    @staticmethod
+    def channel_code():
+        """
+        Returns an capitalized identifier for this channel class, unique among subclasses.
+        """
+        return 'BLACKBOARD'
```

Comment:
This might be the kind of thing that I store in an enum somewhere along with the other channel codes, but, not super important atm.